client/src/components/utils/matomo.tsx

```diff
@@ -0,0 +1,39 @@
+'use client';
+
+import { useState, useEffect } from 'react';
+
+import { usePathname, useSearchParams } from 'next/navigation';
+
+import { init, push } from '@socialgouv/matomo-next';
+
+import env from '@/env.mjs';
+
+const Matomo = () => {
+  const searchParams = useSearchParams();
+  const pathname = usePathname();
+
+  const [initialised, setInitialised] = useState(false);
+
+  // Initialize Matomo
+  useEffect(() => {
+    if (env.NEXT_PUBLIC_MATOMO_URL && env.NEXT_PUBLIC_MATOMO_SITE_ID && !initialised) {
+      init({ url: env.NEXT_PUBLIC_MATOMO_URL, siteId: env.NEXT_PUBLIC_MATOMO_SITE_ID });
+    }
+
+    return () => setInitialised(true);
+  }, [initialised, setInitialised]);
+
+  // Send page views
+  useEffect(() => {
+    if (!pathname) return;
+
+    const url = pathname + (searchParams ? '?' + searchParams.toString() : '');
```

Comment:
Do you think we should send the searchParams? Maybe they could make aggregation of urls difficult as it will have information of the map which maybe is not very interesting 🤔 


"/practices/6024?map-settings=%7B%22boundaries%22%3Atrue%2C%22labels%22%3Atrue%2C%22basemap%22%3A%22basemap-light%22%7D&bbox=%5B-118.13%2C-60.16%2C118.12%2C75%5D&layers=%5B%5D&layers-settings=%7B%7D"

Reply:
I doubted about this as well. Let's send the URLs without the query params then. I'll update the PR.